src/SimpleEventStore/SimpleEventStore.CosmosDb/AzureCosmosDbStorageEngine.cs

```diff
@@ -88,25 +88,39 @@ public async Task<IStorageEngine> Initialise(CancellationToken cancellationToken
                 ? int.MaxValue
                 : startPosition + numberOfEventsToRead;
 
-            var eventsQuery = _collection.GetItemLinqQueryable<CosmosDbStorageEvent>()
-                .Where(x => x.StreamId == streamId && x.EventNumber >= startPosition && x.EventNumber <= endPosition)
-                .OrderBy(x => x.EventNumber)
-                .ToFeedIterator();
-
-            var events = new List<StorageEvent>();
-
-            while (eventsQuery.HasMoreResults)
+            var queryDefinition = new QueryDefinition(@"
+                    SELECT VALUE e
+                    FROM e
+                    WHERE e.streamId = @StreamId
+                        AND (e.eventNumber BETWEEN @LowerBound AND @UpperBound)
+                    ORDER BY e.streamId ASC"
```

Comment:
Order should be by event number

Reply:
Good catch, now how did the tests pass?

Reply:
Agree @afscrome it seems odd that the tests would pass.  It's running under the Cosmos emulator.  Wondering if we've been lucky with this run (not got time ATM to look at the tests) and it's returning it in order, but ideally this should have failed.